docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
+| `status`      | The decryption result status. For a list of possible values and definitions, see [Decryption Response Statuses](#decryption-response-statuses). |
+| `uid`         | The raw UID2 for the corresponding UID2 token.                                                                                                  |
+| `established` | The timestamp indicating when a user first established the UID2 with the publisher.                                                             |
+
+### Decryption Response Statuses
+
+Decryption response codes, and their meanings, are shown in the following table.
+
+| Value                      | Description                                                             |
+|:---------------------------|:------------------------------------------------------------------------|
+| `SUCCESS`                  | The UID2 token was decrypted successfully and a raw UID2 was returned.  |
+| `NOT_AUTHORIZED_FOR_KEY`   | The requester does not have authorization to decrypt this UID2 token.   |
+| `NOT_INITIALIZED`          | The client library is waiting to be initialized.                        |
+| `INVALID_PAYLOAD`          | The incoming UID2 token is not a valid payload.                         |
+| `EXPIRED_TOKEN`            | The incoming UID2 token has expired.                                    |
+| `KEYS_NOT_SYNCED`          | The client has failed to synchronize keys from the UID2 service.        |
+| `VERSION_NOT_SUPPORTED`    | The client library does not support the version of the encrypted token. |
+| `INVALID_TOKEN_LIFETIME`   | The token has invalid timestamp.        |
+
+## Usage for DSPs
+
+
+The following instructions provide an example of how you can decode bid stream tokens using the UID2 SDK for Java as a DSP.
+
+1. Create a `BidstreamClient`:
 
 ```java
-import com.uid2.client.IUID2Client;
- 
-IUID2Client client = UID2ClientFactory.create(TEST_ENDPOINT, TEST_API_KEY, TEST_SECRET_KEY);
-client.refresh(); //Note that refresh() should be called once after create(), and then once per hour
-DecryptionResponse result = client.decrypt(TEST_TOKEN);
+Bidstream client = new BidstreamClient(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);
 ```
 
-### Response Content
+2. Refresh once at startup, and then periodically (recommended refresh interval is hourly):
 
-Available information returned through the SDK is outlined in the following table.
+```java
+client.refresh();
+```
 
-| Function | Description |
-| :--- | :--- |
-| `GetStatus()` | The decryption result status. For a list of possible values and definitions, see [Response Statuses](#response-statuses). |
-| `GetUid()` | The raw UID2 for the corresponding UID2 advertising token. |
-| `GetEstablished()` | The timestamp indicating when a user first established the UID2 with the publisher. |
+3. Decrypt a token into a raw UID2. Pass the domain name of the site where the bid originated from:
```

Comment:
You might want to add [this](https://github.com/IABTechLab/uid2docs/pull/518/files) changes here too

Reply:
Yeah, that makes sense.